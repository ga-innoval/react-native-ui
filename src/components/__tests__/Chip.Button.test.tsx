import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react-native';
import { ChipButton } from '../Chip.Button';

const eventData = {
  nativeEvent: {
    pageX: 20,
    pageY: 30,
  },
};

describe('<ChipButton /> no custom style', () => {
  let component: any = null;
  const mockFn = jest.fn();

  beforeEach(() => {
    component = render(
      <ChipButton text="Press me" iconName="account" onPress={mockFn} />
    );
  });

  it('should render correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should call onPress function', () => {
    const button = screen.getByText(/press/i);
    fireEvent.press(button, eventData);
    expect(mockFn).toBeCalledTimes(1);
  });
});

describe('<ChipButton /> custom style', () => {
  let component: any = null;
  const mockFn = jest.fn();

  beforeEach(() => {
    component = render(
      <ChipButton
        style={{ marginTop: 10 }}
        text="Press me"
        iconName="account"
        onPress={mockFn}
      />
    );
  });

  it('should render correctly', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
