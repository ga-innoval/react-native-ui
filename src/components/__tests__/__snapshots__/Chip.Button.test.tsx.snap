// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChipButton /> custom style should render correctly 1`] = `
<View
  style={
    [
      {
        "alignSelf": "flex-start",
        "borderRadius": 100,
        "overflow": "hidden",
      },
      {
        "marginTop": 10,
      },
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        },
        {
          "backgroundColor": "#232222",
        },
        false,
        false,
      ]
    }
  >
    <Text
      allowFontScaling={false}
      selectable={false}
      style={
        [
          {
            "color": "white",
            "fontSize": 20,
          },
          undefined,
          {
            "fontFamily": "material-community",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      ó°€„
    </Text>
    <Text
      style={
        {
          "color": "white",
          "fontWeight": "bold",
          "marginLeft": 5,
        }
      }
    >
      Press me
    </Text>
  </View>
</View>
`;

exports[`<ChipButton /> no custom style should render correctly 1`] = `
<View
  style={
    [
      {
        "alignSelf": "flex-start",
        "borderRadius": 100,
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "alignSelf": "flex-start",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingHorizontal": 15,
          "paddingVertical": 10,
        },
        {
          "backgroundColor": "#232222",
        },
        false,
        false,
      ]
    }
  >
    <Text />
    <Text
      style={
        {
          "color": "white",
          "fontWeight": "bold",
          "marginLeft": 5,
        }
      }
    >
      Press me
    </Text>
  </View>
</View>
`;
